charts/metal3-deploy/0.1.0/charts/ironic/templates/deployment.yaml

```diff
@@ -23,236 +23,248 @@ spec:
-          {{- $imgname = $.Values.images.ironic.repository }}
-          {{- $imgtag = $.Values.images.ironic.tag }}
-          {{- $imgpullpolicy = $.Values.images.ironic.pullPolicy }}
+      containers:
+      {{- if .Values.global.enable_ironic }}
+      - name: ironic-httpd
+        image: {{ .Values.images.ironic.repository }}:{{ .Values.images.ironic.tag }}
+        imagePullPolicy: {{ .Values.images.ironic.pullPolicy }}
+        command:
+        - /bin/runhttpd
+        envFrom:
+        - configMapRef:
+            name: ironic-bmo
+        {{- if and .Values.requireProvisioningIP (eq $.Values.global.provisioningIP "") }}
+        env:
+        - name: PROVISIONING_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
         {{- end }}
-        - name: {{ .servname }}
-          image: {{ $imgname }}:{{ $imgtag }}
-          imagePullPolicy: {{ $imgpullpolicy }}
-          {{- if .securityContext }}
-          securityContext:
-            {{- toYaml .securityContext | nindent 12 }}
-          {{- end }}
-          {{- if .command }}
-          command:
-            - {{ .command }}
-          {{- end }}
-          {{- if .envFrom }}
-          envFrom:
-          {{- range .envFrom }}
-            - configMapRef:
-                name: {{ .configMapRefname }}
-          {{- end }}
-          {{- end }}
-          {{- if or .env (and .requireProvisioningIP (eq $.Values.global.provisioningIP "")) }}
-          env:
-            {{- range .env }}
-            - name: {{ .name }}
-              valueFrom:
-                {{- if .secretKeyRefKey }}
-                secretKeyRef:
-                  key: {{ .secretKeyRefKey }}
-                  name: {{ .secretKeyRefName }}
-                {{- end }}
-                {{- if .configMapKeyRefKey }}
-                configMapKeyRef:
-                  key: {{ .configMapKeyRefKey }}
-                  name: {{ .configMapKeyRefName }}
-                {{- end }}
-            {{- end }}
-            {{- if and .requireProvisioningIP (eq $.Values.global.provisioningIP "") }}
-            - name: PROVISIONING_IP
-              valueFrom:
-                fieldRef:
-                  fieldPath: status.hostIP
-            {{- end }}
-          {{- end }}
-          {{- if .execCommand }}
-          livenessProbe:
-            exec:
-              command:
-              {{- range .execCommand }}
-                - {{ . }}
-              {{- end }}  
-            failureThreshold: 10
-            initialDelaySeconds: 30
-            periodSeconds: 30
-            successThreshold: 1
-            timeoutSeconds: 10
-          {{- end }}   
-          {{- if .ports }}
-          ports:
-          {{- range .ports }}
-            - containerPort: {{ .containerPort }}
-              name: {{ .name }}
-              protocol: {{ .protocol }}
-          {{- end }}
-          {{- end }}
-          {{- if .execCommand }}
-          readinessProbe:
+        livenessProbe:
```

Comment:
Thanks for doing this refactor - IMO it's going to be much easier to maintain than the previous fully-templated solution! :) 

In future I would suggest proposing large refactor work like this independently of the functional changes though, as it makes it much easier to review.

Reply:
Yes, you are right, I underestimated the amount of changes that I would have to do in order to do the optimisation so I opted to do it in this PR. In retrospect this was not a good idea, sorry about that..